discovery_engine/test/integration/trending_topics_test.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+@Timeout(Duration(seconds: 80))
+
+import 'dart:io' show Directory;
+
+import 'package:test/test.dart';
+import 'package:xayn_discovery_engine/discovery_engine.dart'
+    show TrendingTopicsRequestSucceeded;
+
+import '../logging.dart' show setupLogging;
+import 'utils/helpers.dart'
+    show TestEngineData, initEngine, setupTestEngineData;
+import 'utils/local_newsapi_server.dart' show LocalNewsApiServer;
+
+void main() {
+  setupLogging();
+
+  group(
+    'DiscoveryEngine requestTrendingTopics',
+    () {
+      late LocalNewsApiServer server;
+      late TestEngineData data;
+
+      setUp(() async {
+        data = await setupTestEngineData();
+        server = await LocalNewsApiServer.start();
+      });
+
+      tearDown(() async {
+        await server.close();
+        await Directory(data.applicationDirectoryPath).delete(recursive: true);
+      });
+
+      test('requestTrendingTopics should return trending topics', () async {
```

Comment:
❓ should we also add a test for `TrendingTopicsRequestFailed`, eg like the unresponsive server tests in `request_next_feed_batch_test.dart`? we don't know yet exactly how the DE core implements this, but such a test will likely be useful, maybe with little adjustments.

Reply:
I've added that test, but it's hard test know how this will work before the rust part is finished, so it might probably change.